v1/test/hisi_hpre_test/Makefile.am

```diff
@@ -1,23 +1,23 @@
 AM_CFLAGS=-Wall -O0 -fno-strict-aliasing -I$(top_srcdir) -I$(top_srcdir)/v1/internal -pthread
 
-if WITH_OPENSSL_DIR
-bin_PROGRAMS=test_hisi_hpre hpre_test_tools test_hisi_hpre_times
+if HAVE_CRYPTO
+bin_PROGRAMS=test_hisi_hpre_v1 hpre_test_tools_v1 test_hisi_hpre_times_v1
 
-test_hisi_hpre_SOURCES=test_hisi_hpre.c test_hisi_hpre.h
-hpre_test_tools_SOURCES=hpre_test_tools.c
-test_hisi_hpre_times_SOURCES=test_hisi_hpre_times.c test_hisi_hpre.h hpre_test_sample.h
+test_hisi_hpre_v1_SOURCES=test_hisi_hpre.c test_hisi_hpre.h
+hpre_test_tools_v1_SOURCES=hpre_test_tools.c
+test_hisi_hpre_times_v1_SOURCES=test_hisi_hpre_times.c test_hisi_hpre.h hpre_test_sample.h
 
 if WD_STATIC_DRV
-test_hisi_hpre_LDADD=../../../.libs/libwd.la
-hpre_test_tools_LDADD=../../../.libs/libwd.la
-test_hisi_hpre_times_LDADD=../../../.libs/libwd.la
+test_hisi_hpre_v1_LDADD=../../../.libs/libwd.la
+hpre_test_tools_v1_LDADD=../../../.libs/libwd.la
+test_hisi_hpre_times_v1_LDADD=../../../.libs/libwd.la
 else
-test_hisi_hpre_LDADD=../../../.libs/libwd.so
-hpre_test_tools_LDADD=../../../.libs/libwd.so
-test_hisi_hpre_times_LDADD=../../../.libs/libwd.so
+test_hisi_hpre_v1_LDADD=../../../.libs/libwd.so
+hpre_test_tools_v1_LDADD=../../../.libs/libwd.so
+test_hisi_hpre_times_v1_LDADD=../../../.libs/libwd.so
 endif
 
-test_hisi_hpre_LDADD+=$(with_openssl_dir)/libcrypto_wd.so
-hpre_test_tools_LDADD+=$(with_openssl_dir)/libcrypto_wd.so
-test_hisi_hpre_times_LDADD+=$(with_openssl_dir)/libcrypto_wd.so
+test_hisi_hpre_v1_LDADD+=$(libcrypto_LIBS)
```

Comment:
The original name is "libcrypto_wd.so". When you use libcrypto_LIBS, do you change the name to "libcrypto.so"?

Reply:
Yes, all libcrypto_wd.so changed to libcrypto.so
libcrypto_wd.so is private and can not be built, it is the patch's purpose

Reply:
I noticed that Wenkai Lin changed the name from libcrytpo.so to libcrypto_wd.so. It's better to notify him first. With his  Ack, I could go on.

Reply:
lgtm, the utility of this library is limited and it is not readily available, so it need to be removed. I think we may  disable test directory and add some uadk tool to test_sanity.sh in subsequent patches.